timeseries/src/autogluon/timeseries/models/gluonts/models.py

```diff
@@ -190,6 +191,59 @@ class SimpleFeedForwardModel(AbstractGluonTSModel):
     gluonts_estimator_class: Type[GluonTSEstimator] = SimpleFeedForwardEstimator
 
 
+class TemporalFusionTransformerModel(AbstractGluonTSModel):
+    """TemporalFusionTransformer model for forecasting, as described in [Lim2020]_.
+
+    .. [Lim2021] Lim, Bryan, et al.
+        "Temporal Fusion Transformers for Interpretable Multi-horizon Time Series Forecasting."
+        International Journal of Forecasting. 2021.
+
+    See `AbstractGluonTSModel` for common parameters.
+
+    Other Parameters
+    ----------------
+    context_length : int or None, default = None
+        Number of past values used for prediction.
+        (default: None, in which case context_length = prediction_length)
+    hidden_dim : int, default = 32
+        Size of the hidden layer.
+    num_heads : int, default = 4
+        Number of attention heads in multi-head attention.
+    dropout_rate : float, default = 0.1
+        Dropout regularization parameter
+    batch_size : int, default = 32
+        Size of the mini-batch during training.
+    epochs : int, default = 100
+        Number of epochs the model will be trained for.
+    """
+
+    gluonts_estimator_class: Type[GluonTSEstimator] = TemporalFusionTransformerEstimator
+    supported_quantiles: set = set([0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9])
+
+    def _get_estimator(self) -> GluonTSEstimator:
+        """Return the GluonTS Estimator object for the model"""
+        hyperparameters = self._get_estimator_init_args()
+        # Turning off hybridization prevents MXNet errors when training on GPU
+        hyperparameters["hybridize"] = False
+        # TFT cannot handle arbitrary quantiles, this is a workaround
+        hyperparameters["num_outputs"] = 9
+        if not set(self.quantile_levels).issubset(self.supported_quantiles):
+            raise ValueError(
+                f"{self.name} requires that quantile_levels are a subset of "
+                f"{self.supported_quantiles} (received quantile_levels = {self.quantile_levels})"
+            )
+        with warning_filter():
+            return self.gluonts_estimator_class.from_hyperparameters(**hyperparameters)
+
+    def predict(self, data: TimeSeriesDataFrame, quantile_levels: List[float] = None, **kwargs) -> TimeSeriesDataFrame:
+        if quantile_levels is not None and not set(self.quantile_levels).issubset(self.supported_quantiles):
```

Comment:
is this relying on the fact that `self.quantile_levels` is set by default to [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]? if so, should we make it explicit in the init of this class with something like?

```
def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    self.quantile_levels = self.supported_quantiles  # list not set
```

Reply:
This was a bug that should be fixed now.

- We make sure that `self.quantile_levels` is a subset of `self.supported_quantiles` when initializing the model in `_get_estimator`
- We make sure that if `quantile_levels` are provided as an argument to `predict` (and are not equal to `None`), these are still a subset of `self.supported_quantiles`

The updated code should correctly take care of the second check.